extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/SecuritySupport.java

```diff
@@ -29,4 +42,25 @@ void start() {
         }
     }
 
+    void onClose() {
+        if (onClose != null) {
+            onClose.run();
+        }
+    }
+
+    private static Runnable closeConnectionWhenIdentityExpired(Vertx vertx, WebSocketConnectionImpl connection,
+            SecurityIdentity identity) {
+        if (identity.getAttribute("quarkus.identity.expire-time") instanceof Long expireInSeconds) {
+            long timerId = vertx.setTimer((expireInSeconds * 1000) - System.currentTimeMillis(),
```

Comment:
Why do you subtract the `System.currentTimeMillis()`? I would just use `vertx.setTimer(TimeUnit.SECOND.toMillis(expireInSeconds))`... i.e. run the action after the given time :thinking:.

Reply:
expireInSeconds is a number of seconds from epoch

Reply:
> expireInSeconds is a number of seconds from epoch

yep, the time accepts delay so it needs to be diff

Reply:
> expireInSeconds is a number of seconds from epoch

Ugh, really? So it's not really `expireInSecods` but more like `expireAt`...